rmc/utils/constraint.py

```diff
@@ -955,214 +955,199 @@ def get_rescue_transcripts_and_create_no_breaks_ht(overwrite: bool) -> None:
-            # Thus, increment counter and continue
-            count += 1
-            continue
-        else:
-            temp_ht = get_section_info(
-                ht,
-                section_num=count,
-                section_type="middle",
-                indices=(count - 2, count - 1),
-            )
-            section_ht = section_ht.union(temp_ht)
-            count += 1
-    end_ht = get_section_info(ht, section_num=count, section_type="last", indices=None)
-    return section_ht.union(end_ht)
+    return ((obs_expr - exp_expr) ** 2) / exp_expr
 
 
-def get_unique_transcripts_per_break(
-    ht: hl.Table,
-    max_n_breaks: int,
-) -> Dict[int, Union[Set[str], hl.expr.SetExpression]]:
+def merge_rmc_hts(round_nums: List[int], is_rescue: bool) -> hl.Table:
     """
-    Return the set of transcripts unique to each break number.
-
-    If the set is empty, return an empty SetExpression.
+    Get table of final RMC sections in a given pathway (initial or rescue) after all break searches are complete.
 
-    .. note::
-        - This function will only get unique transcripts for transcripts with one or one + additional breaks.
-        - This will not work for transcripts with two simultaneous breaks.
-        - Assumes input Table is annotated with list containing booleans for whether that locus is a breakpoint
-        (`break_list`).
+    :param round_nums: List of round numbers to merge results across.
+    :param is_rescue: Whether to operate on search in rescue pathway.
 
-    :param ht: Input Table.
-    :param max_n_breaks: Largest number of breaks.
-    :return: Dictionary with break number (key) and set of transcripts unique to that break number or empty SetExpression (value).
+    :return: Table of final RMC sections. Schema:
+        ----------------------------------------
+        Row fields:
+            'section_obs': int64
+            'section_exp': float64
+            'section_oe': float64
+            'section_chisq': float64
+            'search_type': str
+            'transcript': str
+            'interval': interval<locus<GRCh37>>
+        ----------------------------------------
+        Key: ['transcript', 'interval']
+        ----------------------------------------
     """
-    transcripts_per_break = {}
-    # Use `break_list` annotation (list of booleans for whether a row is a breakpoint)
-    # to filter one/additional breaks ht to rows that are significant breakpoints ONLY
-    ht = ht.filter(ht.break_list.any(lambda x: x))
-
-    # Group HT (filtered to breakpoint positions only) by transcript and
-    # count the number of breakpoints associated with each transcript
-    group_ht = ht.group_by("transcript").aggregate(n_breaks=hl.agg.count())
-
-    # Checkpoint to force hail to finish this group by computation
-    group_ht = group_ht.checkpoint(
-        f"{TEMP_PATH}/breaks_per_transcript.ht", overwrite=True
+    if len(round_nums) < 2:
```

Comment:
```suggestion
    logger.warning("This function performs a join followed by a rekey, which will trigger a shuffle!")
    if len(round_nums) < 2:
```
just adding a warning, since this will shuffle and could fail on preemptibles